### PR TITLE
Improve instructions how to use private/local directory

### PR DESCRIPTION
**Pull request by dalanicolai**
Improve instructions how to use private/local directory. This is a new pull request that replaces #13209. 
In this pull request I implemented the suggestions and feedback given in that pull request.
After this one is accepted please close pull request #13209.

### CODE REVIEW

File: private/local/README.md [hunk 1]
```diff
@@ -2,5 +2,34 @@
...
-This is the place to store the local packages that you define in
-the `dotspacemacs-additional-packages` variable of your dotfile.
+This is the place to store the local packages that you define in the
+`dotspacemacs-additional-packages` variable of your dotfile.
+
+Additional packages can be added using the same recipe as for [adding packages
+to layers](https://develop.spacemacs.org/doc/LAYERS.html#packagesel) i.e.:
+
+- For a local package:
+  - load the file explicitly, using the full path to the file, by placing a
```
Comment (duianto) [inline private/local/README.md]:
```suggestion
  - Load the file explicitly, using the full path to the file, by placing a
```
Capitalized the list items first character.

File: private/local/README.md [hunk 2]
```diff
@@ -2,5 +2,34 @@
...
+  `dotspacemacs-additional-packages` within the `dotspacemacs/layers` function
+  of your dotspacemacs file. After placing your package file into this
+  package-directory the file can be loaded, without requiring the full path, by
+  placing a `(require 'package-name)` within the body of the
+  `dotspacemacs/user-config` function of your dotspacemacs file.
+
+- If the package is on (M)ELPA simply add the package name to the list
+  `dotspacemacs-additional-packages` in your dotspacemacs file
+
+- for a package hosted on github the recipe for github packages can be used i.e. add
```
Comment (duianto) [inline private/local/README.md]:
```suggestion
- For a package hosted on github the recipe for github packages can be used i.e. add
```
Capitalized the list items first character.

**Comment by dalanicolai** [comment]
@smile So I don't know how this, works. The PR still shows waiting for response, and Changes requested. But I have processed all suggestions. Guess this PR should be ready now

**Comment by nixmaniack** [comment]
@duianto Those are quite detailed instructions on rebasing using magit 👍 and they look like can be added to wiki for future references.

**Comment by dalanicolai** [comment]
@duianto Thanks for the great instructions! Definitely nice exercise and I guess it prevented some confusion also... I hope;) @nixmaniack @smile13241324 thanks for the reviews... Anyway, did as you said and I guess now the pull request is ready...

**Comment by duianto** [comment]
@nixmaniack Thanks for the suggestion.
I opened a PR:
CONTRIBUTING.org add a link to squash instructions #13420
that adds a link to the comment above. To the end of the line:
>If you still have several commits, squash them into only one commit ([here's a guide](https://github.com/syl20bnr/spacemacs/pull/13381#pullrequestreview-380313280))

https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org#ideally-for-simple-prs-most-of-them

**Comment by smile13241324** [comment]
Looks good will be merged soon.

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs :purple_heart:, I have cherry picked your commit into develop.

**Pull request closed without merging**